tests/Tests/500__jupyterhub/test-jupyterlab-flask-notebook.robot

```diff
@@ -0,0 +1,86 @@
+*** Settings ***
+Resource         ../../Resources/ODS.robot
+Resource         ../../Resources/Common.robot
+Resource         ../../Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
+Library          DebugLibrary
+Suite Setup      Begin Web Test
+Suite Teardown   End Web Test
+
+Resource         ../../Resources/ODS.robot
+Resource         ../../Resources/Common.robot
+Library          DebugLibrary
+Library          JupyterLibrary
+
+
+*** Variables ***
+
+
+*** Test Cases ***
+Open RHODS Dashboard
+  [Tags]  Sanity
+  Wait for RHODS Dashboard to Load
+
+Can Launch Jupyterhub
+  [Tags]  Sanity
+  Launch JupyterHub From RHODS Dashboard Dropdown
+
+Can Login to Jupyterhub
+  [Tags]  Sanity
+  Login To Jupyterhub  ${TEST_USER.USERNAME}  ${TEST_USER.PASSWORD}  ${TEST_USER.AUTH_TYPE}
+  ${authorization_required} =  Is Service Account Authorization Required
+  Run Keyword If  ${authorization_required}  Authorize jupyterhub service account
+  Wait Until Page Contains Element  xpath://span[@id='jupyterhub-logo']
+
+Can Spawn Notebook
+  [Tags]  Sanity
+  Fix Spawner Status
+  Spawn Notebook With Arguments  image=s2i-generic-data-science-notebook
+
+Can Run Flask Test Notebook
+  [Tags]  Sanity
+
+  Wait for JupyterLab Splash Screen  timeout=30
+
+  Maybe Select Kernel
+  ${is_launcher_selected} =  Run Keyword And Return Status  JupyterLab Launcher Tab Is Selected
+  Run Keyword If  not ${is_launcher_selected}  Open JupyterLab Launcher
+  Launch a new JupyterLab Document
+
+  Close Other JupyterLab Tabs
+
+  ##################################################
+  # Git clone repo and run existing notebook
+  ##################################################
+  Navigate Home (Root folder) In JupyterLab Sidebar File Browser
+  Open With JupyterLab Menu  Git  Clone a Repository
+  Input Text  //div[.="Clone a repo"]/../div[contains(@class, "jp-Dialog-body")]//input  https://github.com/tmckayus/flask-notebook
+  Click Element  xpath://div[.="CLONE"]
+
+  # Run the flask server
+  Open With JupyterLab Menu  File  Open from Path…
+  Input Text  //div[.="Open Path"]/../div[contains(@class, "jp-Dialog-body")]//input  flask-notebook/run_flask.ipynb
+  Click Element  xpath://div[.="Open"]
+  Wait Until run_flask.ipynb JupyterLab Tab Is Selected
+  Close Other JupyterLab Tabs
+  Open With JupyterLab Menu  Run  Run All Cells
+  Sleep  10
+
+  # Run the curl command
+  Open With JupyterLab Menu  File  Open from Path…
+  Input Text  //div[.="Open Path"]/../div[contains(@class, "jp-Dialog-body")]//input  flask-notebook/curl_local.ipynb
+  Click Element  xpath://div[.="Open"]
+  Wait Until curl_local.ipynb JupyterLab Tab Is Selected
+  Open With JupyterLab Menu  Run  Run All Cells
+  Wait Until JupyterLab Code Cell Is Not Active
+  ${output} =  Get Text  (//div[contains(@class,"jp-OutputArea-output")])
```

Comment:
This doesn't work. I can't seem to get the framework to read the output from curl in the second notebook, although it displays on screen. Need some help!

Reply:
It just returns an empty string

Reply:
Seems like you are missing the [last()] at the end of the xpath you get text from, this way I think it will get the output from the first code cell in the nb (which might not be the correct one)

Reply:
Since it is a one liner, you could also do 
`Run Cell And Check Output  !curl localhost:5000  {EXPECTED-OUTPUT}`

Reply:
@lugi0 I'll retry  and confirm, I believe I tried both of those things.  Both returned empty string.  I'll check again ....

Reply:
tl;dr For this specific instance add `[last()]` to this line.  

`//div[contains(@class,"jp-OutputArea-output")]` is returning an array since multiple elements match across all open notebooks.  `Get Text` is returning the visible text for the first element which is the hidden notebook on the other tab which causing the empty string

Reply:
I added [last()], the result is the same.

What can replace [last()] ? Is there a way to loop over output results, maybe it's in there somewhere?